drivers/wifi/nrf700x/osal/fw_if/umac_if/src/tx.c

```diff
@@ -872,6 +872,38 @@ unsigned int tx_buff_req_free(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx,
 				/*Adjust the counters*/
 				fmac_dev_ctx->tx_config.outstanding_descs[tx_done_q]--;
 				fmac_dev_ctx->tx_config.outstanding_descs[*ac]++;
+
+				/*Update the queue_map*/
+				switch (desc %
+					(fpriv->num_tx_tokens_per_ac * WIFI_NRF_FMAC_AC_MAX)) {
+					 /* clear the coresponding bit of last access category &
+					  * update the bit for the new access category selected
+					  */
+				case 0:
+					fmac_dev_ctx->tx_config.spare_desc_queue_map &=
+						~(1 << tx_done_q);
+					fmac_dev_ctx->tx_config.spare_desc_queue_map |=
+						(1 << *ac);
+				break;
+				case 1:
+					fmac_dev_ctx->tx_config.spare_desc_queue_map &=
+						~(1 << (4 + tx_done_q));
```

Comment:
Replace magic numbers like 4, 8, 12 below with either a macro or equation (I am assuming the variable suggested in the above comment would help here)?